[Card] Expandable CardText

cgestes opened the pull request:
@chrismcv

hai-cea commented on docs/src/app/components/pages/components/cards.jsx:
```diff
@@ -56,7 +72,25 @@ class CardPage extends React.Component {
+            name: 'expandable',
+            type: 'bool',
+            header: 'optional',
+            desc: 'Wether this card has expandable contents.'
```
Typo here - 'Whether'

hai-cea commented on docs/src/app/components/pages/components/cards.jsx:
```diff
@@ -56,7 +72,25 @@ class CardPage extends React.Component {
+            desc: 'Wether this card has expandable contents.'
+          },
+          {
+            name: 'openExpanded',
```
What do you think about renaming this to `initiallyExpanded`?

hai-cea commented on src/card/card-actions.jsx:
```diff
@@ -1,14 +1,31 @@
+    else if (this.props.expandable === 'close')
+      this.props.onExpanding(true);
+  },
+
```
Do we need PropTypes for these new props?

hai-cea commented on src/card/card-actions.jsx:
```diff
@@ -18,9 +35,20 @@ let CardActions = React.createClass({
+    }
+    if (expandable) {
+      expandable = <IconButton style={styles.expandableIconButton} onClick={this._onExpanding}>{expandable}</IconButton>;
+    }
```
For long JSX lines, I like to do:
```
expandable = (
  <IconButton
    style={styles.expandableIconButton}
    onClick={this._onExpanding}>
    {expandable}
  </IconButton>
);
```

Also, this `onClick` should be `onTouchTap`.

hai-cea commented on src/card/card-actions.jsx:
```diff
@@ -18,9 +35,20 @@ let CardActions = React.createClass({
       });
     });
 
+    var expandable;
```
`let` instead of `var`

cgestes commented:
refactored.

items with expandable={true} will be toggled.
items with expandableController={true} will have an additional child CardExpandable to control the expansion.

better idea welcome...

Would it be a good idea to have something similar for nested list item? (currently if the child is not a ListItem but a component that render a ListItem nested list does not work).

cgestes commented:
Hi @hai-cea

What should we do with this changeset?

Do you think it follows md guideline enough or should it be more restricted?

hai-cea commented:
@cgestes I think it's good. Here's some more feedback:

1. We should use a different icon - please see:
![image](https://cloud.githubusercontent.com/assets/2007468/8654802/59e818c0-2952-11e5-8524-7731960d2e9c.png)

2. Do you think we should rename the `expandableController` prop to `showExpandableButton`?

3. I think the example is a little confusing. I was expecting the top level icon button to expand and collapse the whole card?

cgestes commented:
Hi,

pushed a new batch of commits.

Hope everything is addressed now.

- renamed expandableController prop to showExpandableButton
- changed the icons (really sorry for that)
- updated the example to be more simple and what most people will wants

hai-cea commented:
Looks great - Thanks @cgestes :+1:

cgestes commented:
cool, thank you.
